pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -461,6 +461,24 @@ func (pt PipelineTask) ValidateName() *apis.FieldError {
 // calls the validation routine based on the type of the task
 func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(pt.validateRefOrSpec())
+
+	// Reject cases where APIVersion and/or Kind are specified alongside an embedded Task.
+	// We determine if this is an embedded Task by checking of TaskSpec.TaskSpec.Steps has items.
+	if pt.TaskSpec != nil && len(pt.TaskSpec.TaskSpec.Steps) > 0 {
```

Comment:
If anyone has a better way to determine whether the `PipelineTask` is, in fact, an embedded `Task`, I'd love to hear it. =)

Reply:
We could at least extract this in a function on `PipelineTask`, so that if we change the way to determine it, it won't change that code (and would make that code more readable ? 🙃 )

Reply:
We require today that `kind` is specified if `apiVersion` is, but we don't check the content of `kind` and `apiVersion`.

Should we determine if it's an embedded `Task` or an embedded custom task based on the content of `kind` and `apiVersion`? We could say that (`tekton.dev/*`, `Task`) means it's an embedded `Task` and should be validated accordingly, else it's an embedded custom task and we cannot validate further.

Once we implement PIP, we can add `Pipeline` to the list types that we validate.

Reply:
Also, the current behavior, with custom tasks enabled, with `apiVersion`/`kind` and `TaskSpec.TaskSpec` populated results in a `Run` getting created and never doing anything, so we're not breaking any existing use cases by being strict.